src/Arcus.WebApi.Telemetry.Serilog/Correlation/CorrelationInfoEnricher.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Arcus.WebApi.Correlation;
+using GuardNet;
+using Microsoft.Extensions.DependencyInjection;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Arcus.WebApi.Telemetry.Serilog.Correlation
+{
+    /// <summary>
+    /// Enriches the log events with the correlation information.
+    /// </summary>
+    public class CorrelationInfoEnricher : ILogEventEnricher
+    {
+        private readonly HttpCorrelationInfo _correlationInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CorrelationInfoEnricher"/> class.
+        /// </summary>
+        /// <param name="correlationInfo">The correlation information registered in the <see cref="IServiceCollection"/>.</param>
+        public CorrelationInfoEnricher(HttpCorrelationInfo correlationInfo)
```

Comment:
We need to be careful here, will an instance be created every time or will it only be initialized once?

In other words, will this work correctly at runtime or always re-use correlation info?